### ISSUE TITLE
AR-1566: Running under a prefix does not work at all in 2.1.1

### ISSUE THREAD

**Issue by kloor** [archivesspace/archivesspace#942]
In version 2.1.1 of ArchivesSpace, if the config.rb file contains the settings described in the prefix documentation, ArchivesSpace fails to work correctly and the log file contains the following error:

I, [2017-08-17T11:02:13.069631 #2482]  INFO -- : Started GET "/" for 0:0:0:0:0:0:0:1 at 2017-08-17 11:02:13 -0400
F, [2017-08-17T11:02:13.072080 #2482] FATAL -- :
F, [2017-08-17T11:02:13.072403 #2482] FATAL -- : ActionController::RoutingError (No route matches [GET] "/"):
F, [2017-08-17T11:02:13.072704 #2482] FATAL -- :
F, [2017-08-17T11:02:13.072942 #2482] FATAL -- : config/application.rb:30:in `call'
config/application.rb:30:in `call'

This was tested with a freshly downloaded archive from the releases page, and only changing config.rb as described in: https://github.com/archivesspace/archivesspace/blob/master/README_PREFIX.md

**Comment by jrcwiok** [comment]
I am having the same issue. I upgraded to 2.1.1 hoping it would resolve our proxypass issues. Version 2.1, I could hit the public url that we have the proxypass set up for, but our css, javascripts and fonts would not load. Now I get an error when I go to the Public url.

this is what we have in our config:
AppConfig[:public_proxy_url] = "http://some.library.org/archives/"

   <Location /archives/>

      Allow from all
      AllowOverride None
      #Require all granted

      ProxyPass        http://localhost:8081/
      ProxyPassReverse http://localhost:8081/

   </Location>

**Comment by mark-cooper** [comment]
Running under a prefix is still effectively experimental as of 2.1.1, and to the best of my knowledge is _not_ expected to work. Development effort was put into making prefixes a fully supported option but testing and review was not completed in time, unfortunately.

There's a JIRA about this: https://archivesspace.atlassian.net/browse/AR-1566

**Comment by kloor** [comment]
To be clear: In previous versions, paths in CSS files were broken when running under a prefix. This was fixed in 85a7805, but that was not until after v2.0.1 was released. I'm currently running a v2.0.1 with that change in production.

The new issue is in v2.1.1, and prevents ArchiveSpace from serving pages when on a prefix entirely, as noted by the log snippet I provided above with the fatal errors.

If support for prefixes is experimental, perhaps you should note such in the documentation? Users such as myself assume that since it is documented, it would be supported and not broken.

**Comment by kloor** [comment]
I've installed v2.2.2 running with prefixes, and found that the public interface seems to work including loading all assets. The staff interface partially works, but the CSS files do not load.

I'll gladly provide any additional information, or test any changes to get this working.

**Comment by jrcwiok** [comment]
I've installed v2.2.2, but i'm still having an issue with prefixes working. We only want to use a prefix for the public interface. This is the set up I have:

this is what we have in our config:
AppConfig[:public_proxy_url] = "http://some.library.org/archives/"

in httpd.conf:
<VirtualHost *:80>

   ServerName http://some.library.org:80
<Location /archives/>

  Allow from all
  AllowOverride None
  #Require all granted

  ProxyPass        http://localhost:8081/
  ProxyPassReverse http://localhost:8081/

**Comment by mark-cooper** [comment]
@kloor we believe the prefix issues are resolved for the next release (2.2.3): https://github.com/archivesspace/archivesspace/pull/1097

@jrcwiok there's a reference nginx config that has been tested with prefixes here: https://github.com/archivesspace/archivesspace/blob/master/docker-default.conf

**Comment by AustinTSchaffer** [comment]
The nginx conf has been moved to https://github.com/archivesspace/archivesspace/blob/017c9c6dd6addff844de0e57152b530988e929c0/proxy/default.conf
